### DISCUSSION TITLE
[2023-07-17] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by wdanilo** [enso-org/enso#7307] [Weekly Demo Day]
Based on discussion with Sylwia, James and Ned, we decided that we need to finish mid-term design of the GUI elements. Otherwise, we would not be able to develop it when I will be on vacation or focusing on sales. This week, I was working on designs for cloud and future GUI development, including:
- Managing drive permissions.
- Nodes and files documentation.
- Better templates view.
- Better integration between dashboard and node editor.
- Missing elements (icons, visualization icons, error display, etc).

There are a few screenshots attached below. They are not complete yet – I'm working on writing docs for them and doing final cleanup.

<img width="1444" alt="image" src="https://github.com/enso-org/enso/assets/1623053/57d6c2de-5202-44b4-b0dc-dbac7a9524a3">

<img width="1444" alt="image" src="https://github.com/enso-org/enso/assets/1623053/2823dfb5-cf4f-4183-ac4a-0a781e639844">

<img width="1444" alt="image" src="https://github.com/enso-org/enso/assets/1623053/69c0ecfa-c276-4ef3-aedb-574ac049a7f8">

**Comment by mwu-tow** [comment]
Ongoing work on #7202:
 * replaced `react-hot-toast` with `react-toastify` in the dashboard;
 * mostly integrated `react-toastify` with Rust;
 * some work on packaging.

 Needs few missing pieces of API and code cleanup.

 Further work will likely involve integrating this new notification system with various events in the GUI.

**Comment by farmaazon** [comment]
A New Way of ordering components is still under construction. I did a heavy refactoring and removed many parts of old Component Browser and even older Searcher with Actions etc. The code simplified much.

This screenshot has glitches because I wasn't rebased to newest develop (and rebasing is hard, with many conflicts):

![image](https://github.com/enso-org/enso/assets/3919101/32ea7b24-51b3-466e-b8c6-221eea95b03c)

I need to do some code cleanup and testing, so the PR should be expected soon.

**Comment by vitvakatu** [comment]
Last week I:
- Fixed the regression introduced by me: https://github.com/enso-org/enso/pull/7267
- Fixed an annoying bug/feature which made the "Execution failed" error message persistent: https://github.com/enso-org/enso/pull/7268
- Opened a PR with an updated and reworked IDE top bar presented last demo: https://github.com/enso-org/enso/pull/7249 (waiting on CI)
- Updated and added new icons: https://github.com/enso-org/enso/pull/7296 (waiting on review)
- Working on updating design of the CB: https://github.com/enso-org/enso/issues/7244 (wip)

![icons](https://user-images.githubusercontent.com/6566674/253580216-e3f27e47-04ef-4bdd-9bb0-126936ccb27d.png)

**Comment by kazcw** [comment]
Implemented new search, and scoring API (#7046).

[vokoscreenNG-2023-07-17_07-08-59.webm](https://github.com/enso-org/enso/assets/1047859/7e7c657b-00ed-41a5-86b0-684a1381a675)

Currently testing with a Bookclub, then writing module-level docs before PR.

---

Next:

- Some quick additional CB focus changes.
- Integrate the Enso font.
- #7006

**Comment by MichaelMauderer** [comment]
Worked on https://github.com/enso-org/enso/issues/7199
* updating the breadcrumbs to support an (optional) icon derived from a component browser entry before each items name.
* update the visuals to match the new design (new background, new colours, etc)
* some refactoring in preparation of the breadcrumbs to be re-used in other places

Screenshot showing a demo scene with the breadcrumbs. Top one has dummy content according to the new design, the other on is still "in-place" with the current usage in the component browser.

![image](https://github.com/enso-org/enso/assets/1428930/e8186f2a-8638-464a-8648-98a8c7be715d)

**Comment by Frizi** [comment]
Fixed regression with missing node name labels https://github.com/enso-org/enso/pull/7269
Finished implementation of new node design task, but still reviewing it myself before submitting for review/testing. PR: #7311 

<img width="653" alt="image" src="https://github.com/enso-org/enso/assets/919491/ad4f2eae-5691-456c-82a4-19947e6706e5">
